### PR TITLE
refactor(gatsby-source-contentful): Refactor resolveFixed and resolveFluid

### PR DESCRIPTION
**Pull request by maecapozzi**
## Description

Hi there 👋. I've been working with `gatsby-source-contentful`, and noticed that a lot of functionality was repeated between `resolveFixed` and `resolveFluid`. I thought it might be worth contributing a refactor. 

## Changes
* Split `resolveFixed` and `resolveFluid` out from `extend-node-type.js`
* Break out shared functionality so it can easily be reused across
  both functions
* Extract functionality from `resolveFixed` to improve readability
  and reduce need for comments
* Extract functionality from `resolveFluid` to improve readability
  and reduce need for comments
* Move to ES6 imports/exports

### CODE REVIEW

File: packages/gatsby-source-contentful/.gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 /yarn.lock
 /*.js
 !index.js
+/resolve-fixed-and-fluid/
```
Comment (sidharthachatterjee) [inline packages/gatsby-source-contentful/.gitignore]: Unsure why this is here?

Comment (maecapozzi) [reply packages/gatsby-source-contentful/.gitignore]: Hm...me too! Removing :)

**Comment by maecapozzi** [comment]
@sidharthachatterjee, I think this should be all set now. Let me know if you'd like to me to add anything else.

**Comment by maecapozzi** [comment]
Hey @sidharthachatterjee, I rebased and suddenly I'm getting failures. None of the linting issues are in my code. Should I have avoided rebasing?

**Pull request closed without merging**